### PR TITLE
Add validation helpers

### PR DESCRIPTION
**Pull request by AndreyTretyak**
Add helpers methods for argument validation

### CODE REVIEW

File: src/Extensions/Abstractions/Validation.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		/// <summary>
+		/// Throw exception if value is null, otherwise returns not-nullable value
+		/// </summary>
+		/// <typeparam name="T">type of the value</typeparam>
+		/// <param name="value">value to check</param>
+		/// <param name="name">name of the value for exception message</param>
+		/// <exception cref="ArgumentNullException">if value is null</exception>
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		public static T ThrowIfNull<T>(T? value, string? name = null) where T : class
+			=> value ?? throw new ArgumentNullException(name);
```
Comment (K-Cully) [inline src/Extensions/Abstractions/Validation.cs:23]: what happens here if `name` is null`?

File: tests/Extensions/Abstractions.UnitTests/ValidationTests.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		[TestMethod]
+		public void ThrowIfNullOrWhiteSpace_WhenValueValidRefType_ReturnsIt()
+		{
+			object expected = new object();
+			object actual = Validation.ThrowIfNull(expected);
+			Assert.AreEqual(expected, actual);
+		}
+
+		[TestMethod]
+		public void ThrowIfNullOrWhiteSpace_WhenValueValidStruct_ReturnsIt()
```
Comment (K-Cully) [inline tests/Extensions/Abstractions.UnitTests/ValidationTests.cs:21]: These two tests should be ThrowIfNull_...

File: src/Extensions/Abstractions/Validation.cs (hunk 1 above)
Comment (AndreyTretyak) [reply src/Extensions/Abstractions/Validation.cs:23]: If the name is `null` message would be "Value cannot be null". https://sharplab.io/#v2:C4LgTgrgdgPgAgJgIwFgBQiAM7EGYAEA3uvqfogkSWTcABZgD2A7vlAKasCCYA5hAFt2UYADkIAGwkBRAB4BjdgAdgAS0ZQAFFEkSAlAG5qpAL7oTQA=
If the name is "test" message would be "Value cannot be null. (Parameter 'test')" https://sharplab.io/#v2:C4LgTgrgdgPgAgJgIwFgBQiAM7EGYAEA3uvqfogkSWTcABZgD2A7vlAKasCCYA5hAFt2UYADkIAGwkBRAB4BjdgAdgAS0ZQAFACJg7AM7BtASgDc1UgF90loA===

File: tests/Extensions/Abstractions.UnitTests/ValidationTests.cs (hunk 2 above)
Comment (AndreyTretyak) [reply tests/Extensions/Abstractions.UnitTests/ValidationTests.cs]: Thank you, too much copy/paste

**Pull request merged**